### PR TITLE
refactoring: index file

### LINKED ISSUE #100: Refactoring of index
Please, do refactoring of index.js. Now it have a lot of extra code, move it to parser function.

### PR DESCRIPTION
**Pull request by ifedchankau**
Do refactoring of index file

Closes #100

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -3,11 +3,17 @@
...
-const parser = (help, config) => {
-    config = validate(config);
-    const context = handle(help, config);
-    return templatizer(context, config);
+const parser = (doc, config) => {
+    config = config ? validate(config) : configDefault;
+    if (doc) {
+        const context = handle(doc, config);
+        return templatizer(context, config);
+    } else {
```
Comment (xferra) [inline src/parser.js]: Else block is optional there as far as you always return a value before.

File: src/parser.js [hunk 2]
```diff
@@ -3,11 +3,17 @@
 const handle = require('./handle.js');
 const templatizer = require('./templatizer.js');
 const validate = require('./configValidation.js');
+const configDefault = require('./template/configDefault.json');
+const noDocsError = new Error('No documentation passed');
 
-const parser = (help, config) => {
-    config = validate(config);
```
Comment (xferra) [inline src/parser.js:8]: Assign it to a local variable, functions with side effects are less readable.

Comment (xferra) [reply src/parser.js:8]: It also confuses that `validate` checks and modify the object.

File: src/parser.js [hunk 3]
```diff
@@ -3,11 +3,17 @@
...
-    const context = handle(help, config);
```
Comment (xferra) [inline src/parser.js:9]: `Handle` is too generic and without docs gives no idea what it handles.

File: src/parser.js (hunk 2 above)
Comment (ifedchankau) [reply src/parser.js:8]: @xferra done

File: src/parser.js (hunk 3 above)
Comment (ifedchankau) [reply src/parser.js:9]: @xferra done

**Review by xferra — APPROVED** [review]
+1

**Pull request merged**